lua/lualine/utils.lua

```diff
@@ -18,4 +18,303 @@ function M.draw_section(section, separator)
+  {'183', { 215, 175, 255 }},
+  {'184', { 215, 215,   0 }},
+  {'185', { 215, 215,  95 }},
+  {'186', { 215, 215, 135 }},
+  {'187', { 215, 215, 175 }},
+  {'188', { 215, 215, 215 }},
+  {'189', { 215, 215, 255 }},
+  {'190', { 215, 255,   0 }},
+  {'191', { 215, 255,  95 }},
+  {'192', { 215, 255, 135 }},
+  {'193', { 215, 255, 175 }},
+  {'194', { 215, 255, 215 }},
+  {'195', { 215, 255, 255 }},
+  {'196', { 255,   0,   0 }},
+  {'197', { 255,   0,  95 }},
+  {'198', { 255,   0, 135 }},
+  {'199', { 255,   0, 175 }},
+  {'200', { 255,   0, 215 }},
+  {'201', { 255,   0, 255 }},
+  {'202', { 255,  95,   0 }},
+  {'203', { 255,  95,  95 }},
+  {'204', { 255,  95, 135 }},
+  {'205', { 255,  95, 175 }},
+  {'206', { 255,  95, 215 }},
+  {'207', { 255,  95, 255 }},
+  {'208', { 255, 135,   0 }},
+  {'209', { 255, 135,  95 }},
+  {'210', { 255, 135, 135 }},
+  {'211', { 255, 135, 175 }},
+  {'212', { 255, 135, 215 }},
+  {'213', { 255, 135, 255 }},
+  {'214', { 255, 175,   0 }},
+  {'215', { 255, 175,  95 }},
+  {'216', { 255, 175, 135 }},
+  {'217', { 255, 175, 175 }},
+  {'218', { 255, 175, 215 }},
+  {'219', { 255, 175, 255 }},
+  {'220', { 255, 215,   0 }},
+  {'221', { 255, 215,  95 }},
+  {'222', { 255, 215, 135 }},
+  {'223', { 255, 215, 175 }},
+  {'224', { 255, 215, 215 }},
+  {'225', { 255, 215, 255 }},
+  {'226', { 255, 255,   0 }},
+  {'227', { 255, 255,  95 }},
+  {'228', { 255, 255, 135 }},
+  {'229', { 255, 255, 175 }},
+  {'230', { 255, 255, 215 }},
+  {'231', { 255, 255, 255 }},
+
+  -- Gray-scale range.
+  {'232', {   8,   8,   8 }},
+  {'233', {  18,  18,  18 }},
+  {'234', {  28,  28,  28 }},
+  {'235', {  38,  38,  38 }},
+  {'236', {  48,  48,  48 }},
+  {'237', {  58,  58,  58 }},
+  {'238', {  68,  68,  68 }},
+  {'239', {  78,  78,  78 }},
+  {'240', {  88,  88,  88 }},
+  {'241', {  98,  98,  98 }},
+  {'242', { 108, 108, 108 }},
+  {'243', { 118, 118, 118 }},
+  {'244', { 128, 128, 128 }},
+  {'245', { 138, 138, 138 }},
+  {'246', { 148, 148, 148 }},
+  {'247', { 158, 158, 158 }},
+  {'248', { 168, 168, 168 }},
+  {'249', { 178, 178, 178 }},
+  {'250', { 188, 188, 188 }},
+  {'251', { 198, 198, 198 }},
+  {'252', { 208, 208, 208 }},
+  {'253', { 218, 218, 218 }},
+  {'254', { 228, 228, 228 }},
+  {'255', { 238, 238, 238 }},
+}
+
+local color_table_meta = {
+  __index = function(color_table, hex_color)
+    local function get_color_deviation(color1, color2)
+      -- returns how much color2 deviates from color1
+      local dr = math.abs(color1[1] - color2[1]) / (color1[1]+1) * 100
+      local dg = math.abs(color1[2] - color2[2]) / (color1[2]+1) * 100
+      local db = math.abs(color1[3] - color2[3]) / (color1[3]+1) * 100
+      return (dr + dg + db)
+    end
+
+    local r = tonumber(hex_color:sub(2,3), 16)
+    local g = tonumber(hex_color:sub(4,5), 16)
+    local b = tonumber(hex_color:sub(6,7), 16)
+
+    -- check which color deviates lest from haxcolor
+    local lest_deviation = {0, 10000}
```

Comment:
This algorithm could probably be improved in time complexity, I will take a look at it later.